pkg/cloud/azure/services/virtualmachines/service.go

```diff
@@ -30,8 +30,8 @@ type Service struct {
 }
 
 // getVirtualNetworksClient creates a new groups client from subscriptionid.
-func getVirtualMachinesClient(subscriptionID string, authorizer autorest.Authorizer) compute.VirtualMachinesClient {
-	vmClient := compute.NewVirtualMachinesClient(subscriptionID)
+func getVirtualMachinesClient(resourceManagerEndpoint, subscriptionID string, authorizer autorest.Authorizer) compute.VirtualMachinesClient {
+	vmClient := compute.NewVirtualMachinesClientWithBaseURI(resourceManagerEndpoint, subscriptionID)
```

Comment:
This was just one example of a place where the base URI needs to be specified when creating the client. The "NewXXXClient" calls in each of the services need to be changes to "NewXXXClientWithBaseURI".

Reply:
👍